python/omnisharp/commands.py

```diff
@@ -0,0 +1,393 @@
+    if version == 'v1':
+        endpoint = '/getcodeactions'
+    elif version == 'v2':
+        endpoint = '/v2/getcodeactions'
+    response = getResponse(ctx, endpoint, parameters, json=True)
+    return response['CodeActions']
+
+
+@vimcmd
+def runCodeAction(mode, action, version='v1'):
+    parameters = codeActionParameters(mode, version)
+    if version == 'v1':
+        parameters['codeaction'] = action
+        response = getResponse(ctx, '/runcodeaction', parameters, json=True)
+        if 'Text' in response:
+            setBuffer(response['Text'])
+            return True
+    elif version == 'v2':
+        parameters['identifier'] = action
+        response = getResponse(ctx, '/v2/runcodeaction', parameters, json=True)
+        changes = response.get('Changes')
+        if changes:
+            bufname = vim.current.buffer.name
+            pos = vim.current.window.cursor
+            for changeDefinition in changes:
+                filename = formatPathForClient(ctx,
+                                               changeDefinition['FileName'])
+                openFile(filename, noautocmd=1)
+                if not setBuffer(changeDefinition.get('Buffer')):
+                    for change in changeDefinition.get('Changes', []):
+                        setBuffer(change.get('NewText'))
+            openFile(bufname, pos[0], pos[1], 1)
+            return True
+    return False
+
+
+def codeActionParameters(mode, version='v1'):
+    parameters = {}
+    if mode == 'visual':
+        start = vim.eval('getpos("\'<")')
+        end = vim.eval('getpos("\'>")')
+        if version == 'v1':
+            parameters['SelectionStartLine'] = start[1]
+            parameters['SelectionStartColumn'] = start[2]
+            parameters['SelectionEndLine'] = end[1]
+            parameters['SelectionEndColumn'] = end[2]
+        elif version == 'v2':
+            parameters['Selection'] = {
+                'Start': {'Line': start[1], 'Column': start[2]},
+                'End': {'Line': end[1], 'Column': end[2]}
+            }
+    return parameters
+
+
+@vimcmd
+def fixCodeIssue():
+    response = getResponse(ctx, '/fixcodeissue', json=True)
+    setBuffer(response.get('Text'))
+
+
+@vimcmd
+def getCodeIssues():
+    response = getResponse(ctx, '/getcodeissues', json=True)
+    return quickfixes_from_response(ctx, response['QuickFixes'])
+
+
+@vimcmd
+def codeCheck():
+    response = getResponse(ctx, '/codecheck', json=True)
+    return quickfixes_from_response(ctx, response['QuickFixes'])
+
+
+@vimcmd
+def signatureHelp():
+    return getResponse(ctx, '/signatureHelp', json=True)
+
+
+@vimcmd
+def typeLookup(include_documentation):
+    parameters = {
+        'includeDocumentation': bool(include_documentation),
+    }
+    response = getResponse(ctx, '/typelookup', parameters, json=True)
+    return {
+        'type': response.get('Type', ''),
+        'doc': response.get('Documentation', ''),
```

Comment:
When checking type on whitespace or builtins etc., the server returns `{"Type":null,"Documentation":null}`, and `response.get` returns `None` for these, which the vim calling functions don't like. This works:

```python
    type = response.get('Type', '')
    doc = response.get('Documentation', '')
    return {
        'type': '' if type is None else type,
        'doc': '' if doc is None else doc
    }
```

Reply:
Good catch! If you don't mind, I'll do the lazy-man's check 
```python
response.get('Type', '') or ''
```
😛